src/firmware/app/control/trajectory_planner.c

```diff
@@ -173,36 +178,73 @@ void app_trajectory_planner_interpolateConstantTimeTrajectory(
             const float delta_y =
                 variable_time_trajectory->trajectory_elements[i].position.y -
                 variable_time_trajectory->trajectory_elements[i - 1].position.y;
+            const float delta_speed = variable_time_trajectory->speed_profile[i] -
+                                      variable_time_trajectory->speed_profile[i - 1];
 
-            const float slope_x = delta_x / delta_time;
-            const float slope_y = delta_y / delta_time;
+            const float slope_x     = delta_x / delta_time;
+            const float slope_y     = delta_y / delta_time;
+            const float slope_speed = delta_speed / delta_time;
 
             const float interpolated_x =
                 slope_x * (trajectory_time -
                            variable_time_trajectory->trajectory_elements[i - 1].time) +
                 variable_time_trajectory->trajectory_elements[i - 1].position.x;
+
             const float interpolated_y =
                 slope_y * (trajectory_time -
                            variable_time_trajectory->trajectory_elements[i - 1].time) +
                 variable_time_trajectory->trajectory_elements[i - 1].position.y;
 
+            const float interpolated_speed =
+                slope_speed *
+                    (trajectory_time -
+                     variable_time_trajectory->trajectory_elements[i - 1].time) +
+                variable_time_trajectory->speed_profile[i - 1];
+
             constant_period_trajectory->trajectory_elements[time_periods].position.x =
                 interpolated_x;
             constant_period_trajectory->trajectory_elements[time_periods].position.y =
                 interpolated_y;
+            constant_period_trajectory->speed_profile[time_periods] = interpolated_speed;
             constant_period_trajectory->trajectory_elements[time_periods].time =
                 trajectory_time;
 
+            // Step forwards one interpolation period
             time_periods++;
             trajectory_time = interpolation_period * time_periods;
         }
     }
 
-    constant_period_trajectory->num_elements = time_periods;
+    if (time_periods == TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS)
+    {
+        return INTERPOLATION_ELEMENT_MAXED_OUT;
+    }
+    else
+    {
+        // The last element of both trajectories are also identical
+        constant_period_trajectory->trajectory_elements[time_periods].time =
+            variable_time_trajectory
+                ->trajectory_elements
+                    [variable_time_trajectory->path_parameters.num_segments - 1]
```

Comment:
Can we pull `variable_time_trajectory->path_parameters.num_segments - 1` out as a local variable?

Reply:
Pulled out into a local 'last element_index'